src/mappings/Rewards.ts

```diff
@@ -310,3 +311,48 @@ async function updateAccumulatedReward(event: SubstrateEvent, isReward: boolean)
     accumulatedReward.amount = accumulatedReward.amount + (isReward ? newAmount : -newAmount)
     await accumulatedReward.save()
 }
+
+
+async function handleParachainRewardForTxHistory(rewardEvent: SubstrateEvent): Promise<void> {
+    const extrinsic = rewardEvent.extrinsic;
+    const block = rewardEvent.block;
+    const blockNumber = block.block.header.number.toString()
+    const blockTimestamp = timestamp(block)
+    const {event: {data: [account, amount]}} = rewardEvent
+    const eventId = eventIdFromBlockAndIdx(blockNumber, rewardEvent.idx.toString())
+    const era = await api.query.parachainStaking.round();
+    const eraIndex = JSON.parse(JSON.stringify(era.toJSON())).current
```

Comment:
Why do we need this conversion? If it is a workaround for something, I suggest leaving a comment